flag/flags.go

```diff
@@ -0,0 +1,12 @@
+package flag
+
+// Flags provides the ability to create and communicate feature flags.
+type Flags uint64
+
+// Flags is a bitset
+const (
+	Debug Flags = 1 << iota
+	SamplingSet
```

Comment:
I'd like to clarify a bit why serializing optionality makes less sense in
formats like thrift or protobuf. If took me a while to understand all this
stuff, so figured it might be helpful before things go to cement.

By fixed-width, I meant something like finagle's serialization format. It
is 32bytes in length regardless of if it is a parent or child or flags are
set, and cannot be longer. This format has no keywords like thrift or
protobuf do to indicate optionatlity, hence the need for flags.
FixedWidth/TraceIdCarrier is the one that needs the special understanding
of optional fields, as well special understanding that it by definition
cannot support propagating random baggage.

protobuf isn't fixed width unless all fields are required and fixed length,
and that proto isn't extended by anyone. For example, if a proto includes a
map field, it isn't fixed width, as any entry will change the serialized
length. Since proto is variable length already, and the format has a means
to indicate optionality, it could make more sense to just make the
parent_id optional in the proto. this would be more idiomatic both in
format and also in use.